packages/vue/src/components/IonRouterOutlet.ts

```diff
@@ -374,22 +376,22 @@ export const IonRouterOutlet = defineComponent({
         /**
          * IonRouterOutlet does not support named outlets.
          */
-        if (c.matchedRoute?.props?.default) {
-          const matchedRoute = c.matchedRoute;
-          const routePropsOption = matchedRoute.props.default;
-          const routeProps = routePropsOption
-            ? routePropsOption === true
-              ? c.params
-              : typeof routePropsOption === 'function'
-              ? routePropsOption(matchedRoute)
-              : routePropsOption
-            : null
-
-          props = {
-            ...props,
-            ...routeProps
-          }
+        const routePropsOption = c.matchedRoute?.props?.default;
+        const routeProps = routePropsOption
+          ? routePropsOption === true
+            ? c.params
+            : typeof routePropsOption === 'function'
+
+            // Only call props function on initial render
+            ? !c.ionPageElement && routePropsOption(injectedRoute)
```

Comment:
This isn't quite right. If you navigate from a page that has a props function then go back, the props will be cleared.

Reply:
Two options here are:

1. Allow the props fn to be called whenever the component is rendered.
2. Cache the results of the props function in the view stack item.

Reply:
I think it would be annoying to have the function called every time. Leaning towards option 2.